llvm/lib/CodeGen/PHIElimination.cpp

```diff
@@ -136,6 +136,7 @@ INITIALIZE_PASS_END(PHIElimination, DEBUG_TYPE,
 
 void PHIElimination::getAnalysisUsage(AnalysisUsage &AU) const {
   AU.addUsedIfAvailable<LiveVariables>();
+  AU.addUsedIfAvailable<LiveIntervals>();
```

Comment:
Normally you would only add this if PHIElimination can do a better job when LiveIntervals is available. If you are just saying that PHIElimination is capable of updating LiveIntervals then you should not add this.

In this case it probably won't make any difference, but in the general case, not adding unnecessary "used if available" dependencies will allow the legacy pass manager to free unneeded analyses sooner, possibly saving some memory.

Reply:
It seemed to me that if we use an analysis then we should declare that usage.
But I accept your logic and it doesn't make a different in practice here.